csharp/ql/lib/semmle/code/csharp/dataflow/FlowSummary.qll

```diff
@@ -155,3 +154,45 @@ private class RecordConstructorFlowRequiredSummaryComponentStack extends Require
     )
   }
 }
+
+class Provenance = Impl::Public::Provenance;
+
+private import semmle.code.csharp.frameworks.system.linq.Expressions
+
+private SummaryComponent delegateSelf() {
+  exists(ArgumentPosition pos |
+    result = SummaryComponentInternal::parameter(pos) and
+    pos.isDelegateSelf()
+  )
+}
+
+private predicate mayInvokeCallback(Callable c, int n) {
+  c.getParameter(n).getType() instanceof SystemLinqExpressions::DelegateExtType and
+  not c.fromSource()
+}
+
+private class SummarizedCallableWithCallback extends SummarizedCallable {
```

Comment:
I believe I added the equivalent of this for Java in relation to variable capture flow. Curious what role it plays in this PR? I don't mind adding it here, it's just that it looks more related to capture rather than the lambda dispatch precision improvement.

Reply:
Indeed, this is only needed for captured variables, which I realized after having written it. But it can't hurt to add now.